tests/e2e/util.go

```diff
@@ -2269,3 +2271,58 @@ func getDefaultDatastore(ctx context.Context) *object.Datastore {
 
 	return defaultDatastore
 }
+
+//setClusterDistributionValue returns boolean
+func setClusterDistributionValue(ctx context.Context, client clientset.Interface, clusterDistribution string) bool {
+	framework.Logf("Inside the setClusterDistributionValue function, and value to set is = %s", clusterDistribution)
+	isClusterDistributionValueSet := false
+
+	// Get the current cluster-distribution value from secret
+	currentSecret, err := client.CoreV1().Secrets(kubeSystemNamespace).Get(ctx, configSecret, metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	framework.Logf("Secret Name is %s\n", currentSecret.Name)
+
+	//Read and map the content of csi-vsphere.conf to a variable
+	originalConf := string(currentSecret.Data[vSphereCSIConf])
+	cfg, err := readConfigFromSecretString(originalConf)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	//Current value of cluster-distribution is
+	framework.Logf("Cluster-distribution value before modifying is = %s\n", cfg.Global.ClusterDistribution)
+
+	// Check if the cluster-distribution value is as required or reset
+	if cfg.Global.ClusterDistribution != clusterDistribution {
+		// Modify csi-vsphere.conf file
+		modifiedConf := fmt.Sprintf("[Global]\ninsecure-flag = \"%t\"\ncluster-id = \"%s\"\ncluster-distribution = \"%s\"\n\n[VirtualCenter \"%s\"]\nuser = \"%s\"\npassword = \"%s\"\ndatacenters = \"%s\"\nport = \"%s\"\n",
+			cfg.Global.InsecureFlag, cfg.Global.ClusterID, clusterDistribution, cfg.Global.VCenterHostname, cfg.Global.User, cfg.Global.Password, cfg.Global.Datacenters, cfg.Global.VCenterPort)
+
+		// Set modified csi-vsphere.conf file and update
+		currentSecret.Data[vSphereCSIConf] = []byte(modifiedConf)
+		_, err := client.CoreV1().Secrets(kubeSystemNamespace).Update(ctx, currentSecret, metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		framework.Logf("Sleeping for volume health to reflect the cluster-distribution value = %s\n", clusterDistribution)
+		time.Sleep(time.Duration(healthStatusWaitTime))
+
+	} else {
```

Comment:
we can also do away with the else block

Reply:
If else is removed, the get secret API will be called twice. So, I am breaking out of the function in else

Reply:
let try this flow

if(distribution does not match) {
        change the distribution
        verify the change
}
return isClusterDistributionValueSet